fedbadges/utils.py

```diff
@@ -111,13 +112,29 @@ def notification_callback(topic, msg):
 def user_exists_in_fas(config, user):
     """ Return true if the user exists in FAS. """
     default_url = 'https://admin.fedoraproject.org/accounts/'
+    if config.get["fasjson"]:
+        return user_exists_in_freeipa_fas(config, user)
+
     fas2 = fedora.client.AccountSystem(
         base_url=config['fas_credentials'].get('base_url', default_url),
         username=config['fas_credentials']['username'],
         password=config['fas_credentials']['password'],
     )
     return bool(fas2.person_by_username(user))
 
+def user_exists_in_fasjson(config, user):
+    """ Return true if the user exists in freeipa-fas. """
+    default_url = 'https://admin.fedoraproject.org/accounts/'
+    client = Client(
+        url=config['fas_credentials'].get('base_url', default_url)
+    )
+    try:
+        user = client.get_user(username=user).result
+        return True
+    except fasjson_client.errors.APIError:
```

Comment:
I am a bit unclear on that part. APIError seems to cover all type of error, from user that do not exist to error 500 or anything. So, if my understanding is correct, should we log some of those errors ?

Reply:
I guess we can, although I'm not 100% how many times this function gets called on average - if its a lot then potentially we could really pollute the logs by logging every time a user is not found. 

I've added it now anyway so let me know what you think.

Reply:
In the current version, all errors beside HTTP 404 get logged. Is this okay, @mscherer?

Reply:
Ahh, didn't spot the discussion below. Considering this as resolved then.